app/main.py

```diff
@@ -1,3 +1,65 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
```

Comment:
Similar to `__add__`, the `__iadd__` method should handle cases where `other` is neither a `Distance` instance nor a number by raising an exception.